src/Microsoft.OData.Edm/ExtensionMethods/TargetPathHelper.cs

```diff
@@ -0,0 +1,232 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="TargetPathHelper.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.Edm
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Microsoft.OData.Edm.Vocabularies;
+
+    /// <summary>
+    /// Target path helper.
+    /// </summary>
+    internal static class TargetPathHelper
+    {
+        /// <summary>
+        /// Get the Segments from a target path string as enumeration of Edm elements.
+        /// </summary>
+        /// <param name="model">The EdmModel.</param>
+        /// <param name="targetSegments">Segments for the target path..</param>
+        /// <param name="caseInsensitive">Property name case-insensitive or not.</param>
+        /// <returns>The created enumeration of <see cref="IEdmElement"/>.</returns>
+        public static IEnumerable<IEdmElement> GetTargetSegments(this IEdmModel model, string[] targetSegments, bool caseInsensitive)
+        {
+            IEdmEntityContainer container = model.FindEntityContainer(targetSegments[0]);
+            List<IEdmElement> pathSegments = new List<IEdmElement>();

```

Comment:
>List<IEdmElement> pathSegments = new List<IEdmElement>(); [](http://example.com/codeflow?start=12&length=57)

it's better to move this line after the following `if (...)` clause, so, we don't need to construct this list if it's not starting from container.